internal/wasm/sdk/opa/vm.go

```diff
@@ -170,8 +170,14 @@ func newVM(policy []byte, data []byte, memoryMin, memoryMax uint32) (*vm, error)
 	return v, nil
 }
 
-func (i *vm) Eval(ctx context.Context, entrypoint EntrypointID, input *interface{}) (interface{}, error) {
-	if err := i.setHeapState(i.evalHeapPtr); err != nil {
+func (i *vm) Eval(ctx context.Context, entrypoint EntrypointID, input *interface{}, metrics metrics.Metrics) (interface{}, error) {
+	metrics.Timer("opa_wasm_vm_eval_total").Start()
+	defer metrics.Timer("opa_wasm_vm_eval_total").Stop()
+
+	metrics.Timer("opa_wasm_vm_eval_set_heap_state").Start()
+	err := i.setHeapState(i.evalHeapPtr)
+	metrics.Timer("opa_wasm_vm_eval_set_heap_state").Stop()
```

Comment:
nit: I'd like to just see this bundled into the "input preparation" metric since we already have several calls into wasm land and this is just one more.